server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/EnqueuedMailsDAO.java

```diff
@@ -165,15 +166,16 @@ Mono<Void> insert(EnqueuedItemWithSlicingContext enqueuedItemWithSlicing) {
             .setMap(ATTRIBUTES, toRawAttributeMap(mail), String.class, ByteBuffer.class)
             .setList(PER_RECIPIENT_SPECIFIC_HEADERS, toTupleList(userHeaderNameHeaderValueTriple, mail.getPerRecipientSpecificHeaders()), TupleValue.class);
 
+        AtomicReference<BoundStatement> finalBoundStatement = new AtomicReference<>(statement);
```

Comment:
Atomic references have a meaning in concurrency related code, and implies a memory barrier that goes and hit RAM causing L1 L2 and L3 cache misses. This brings a serious performance hit. We shall use that only when necessary.

How about cherry-picking https://github.com/quantranhong1999/james-project/commit/75729c37215c1cad1c565d46ec8682330b1b4fb1